### ISSUE TITLE
关于复现 DuetGraph 实验结果的一个问题咨询

### ISSUE THREAD

**Issue by lh5533223** [USTC-DataDarknessLab/DuetGraph#4]
尊敬的作者您好，我目前正在复现您论文《DuetGraph: Coarse-to-Fine Knowledge Graph
Reasoning with Dual-Pathway Global-Local Fusion》的相关实验，非常感谢您分享如此有价值的工作以及开源的代码实现。
        但是我在按照论文描述和官方代码配置进行复现的过程中，我在FB15k237和WN18RR数据集（Transductive和Inductive）上复现的实验结果（mrr、hits@1和hits@10）都要较低于论文中报告的实验结果。因此，想向您进一步请教实验的参数设置、运行环境以及 .sh 脚本的具体复现流程，方便我更准确地对齐实验设置。非常感谢！

**Comment by lijinustc** [comment]
Could you please share your experiment script, the corresponding parameter settings, and your reproduction results?

**Comment by lh5533223** [comment]
好的，就举例在inductive设置下的FB15k v1版本数据集上的复现结果来说，我把DuetGraph-main主目录下的run_fb15k_v1.sh和dual_pathway中的run_fb15k_v1.sh脚本都运行了一遍，脚本运行参数如下：
python main.py --seed 42 \
               --accelerator gpu \
               --strategy ddp \
               --precision 32 \
               --devices 4 \
               --max_epochs 20 \
               --checkpoint_save_path ./experiments/train/fb15k-237_v1/ \
               --data_path ./data/inductive/fb15k-237_v1 \
               --batch_size 64 \
               --test_batch_size 64 \
               --num_workers 8 \
               --num_layer 3 \
               --num_qk_layer 2 \
               --num_v_layer 3 \
               --hidden_dim 64 \
               --num_heads 4 \
               --loss_fn bce \
               --adversarial_temperature 0.5 \
               --num_negative_sample 6 \
               --learning_rate 5e-3 \
               --optimizer Adam \
               --weight_decay 1e-4 \
               --k 4 \
               --delta 8 \
               --dataset fb237_v1
我复现的结果如下所示：
Test results : {'test_mrr': tensor(0.4600, device='cuda:0'), 'test_hits1': tensor(0.3707, device='cuda:0'), 'test_hits10': tensor(0.6049, device='cuda:0')}
还请您过目，非常感谢！

**Comment by lh5533223** [comment]
我的GPU的话是一张NVIDIA GeForce RTX 3090

**Comment by lijinustc** [comment]
According to your results, it appears that you only ran the dual_pathway module without enabling the two-stage inference. In addition, your hyperparameter settings are inconsistent with those reported in the paper. For example, we use a batch size of 96 in our experiments.

**Comment by lh5533223** [comment]
好的好的，十分感谢。超参数设置的话我直接是用的源码中默认的参数并没有去修改，那我这边要对照论文中的附录超参数信息进一步调节一下是吗？
另外还有这个启用两阶段推理的运行，是直接运行DuetGraph-main主目录的run_fb15k_v1.sh脚本文件吗？脚本文件内容如下：
#!/bin/bash


cd dual_pathway
bash run_fb15k_v1.sh > ../dual_pathway_fb15k_v1.log 2>&1 &
dual_pathway_PID=$!
cd ..


cd RED-GNN/inductive
bash run_fb15k_v1.sh > ../../redgnn.log 2>&1 &
REDGNN_PID=$!
cd ../..


wait $dual_pathway_PID
wait $REDGNN_PID

echo "look log."
然后我这边得到的fb15k的日志内容如上传文件所示，感觉跟原文中的实验效果也有一点差距，再次感谢您的回复！

[dual_pathway_fb15k_v1.log](https://github.com/user-attachments/files/24334859/dual_pathway_fb15k_v1.log)

**Comment by lijinustc** [comment]
You can simply run the script in the root directory. Based on the uploaded log files, the results still appear to be from running the dual_pathway script.

**Comment by lh5533223** [comment]
好的，谢谢您的回复。我这边已经解决好这个问题了，目前已经实现了two-stage推理，感谢您！

**Comment by Vincent-swt** [comment]
尊敬的作者您好，我目前在复现您论文《DuetGraph: Coarse-to-Fine Knowledge Graph Reasoning with Dual-Pathway Global-Local Fusion》的相关实验，非常感谢您分享如此有价值的工作以及开源的代码实现。
已经到Figure 3，对于Figure 3 部分我想询问，作者是否将这几个best models in each category的baselines：AdaProp (message passing-based), SAttLE (transformer-based), and KnowFormer (hybrid)的实验也都复现了,其中SAttLE貌似需要很长的时间。因此，想向您进一步请教该部分实验的具体复现流程，非常感谢！

**Comment by lijinustc** [comment]
> ```
> 尊敬的作者您好，我目前在复现您论文《DuetGraph: Coarse-to-Fine Knowledge Graph Reasoning with Dual-Pathway Global-Local Fusion》的相关实验，非常感谢您分享如此有价值的工作以及开源的代码实现。
> ```
> 
> 已经到Figure 3，对于Figure 3 部分我想询问，作者是否将这几个best models in each category的baselines：AdaProp (message passing-based), SAttLE (transformer-based), and KnowFormer (hybrid)的实验也都复现了,其中SAttLE貌似需要很长的时间。因此，想向您进一步请教该部分实验的具体复现流程，非常感谢！

Hello, thank you very much for your interest! Regarding the three baselines you mentioned (AdaProp, SAttLE, and KnowFormer), we have reproduced all of them in the experiments for Figure 3. As you noticed, SAttLE indeed requires a relatively long training time, which is consistent with what is shown in Figure 3.

To ensure a fair comparison, we strictly followed the official commands provided in each baseline’s GitHub repository during reproduction, and we kept key hyperparameters such as batch size and number of layers consistent across all evaluations. This ensures that all baselines were assessed under the same conditions as our model.
